style_guide.md

```diff
@@ -0,0 +1,308 @@
+# Jsonnet Style Guide
+This style guide documents how we want to handle Jsonnet files when building the schemas with consistency. For more general principles and styling rules go to [Databricks repository](https://github.com/databricks/jsonnet-style-guide#databricks-jsonnet-guide).
+
+## Folder structure
+- The following structure is used:
+
+    ```
+    england-wales/
+       lib/
+          common_rules.libsonnet
+          placeholders.libsonnet
+       <questionnaire>/
+          <questionniare>.jsonnet
+          lib/
+             rules.libsonnet
+          blocks/
+             <section-one>/
+                <block-one>.jsonnet
+                <block-two>.jsonnet
+             <section-two>/
+    northern-ireland/   
+    ```
+- Jsonnet source files are separated into two top-level folders - `england-wales` and `northern-ireland` 
+- The `lib/` folder contains common Jsonnet used across all questionnaire schemas
+  - The file extension should be `.libsonnet`
+  - Each file is grouped by function, currently `common_rules` and `placeholders`
+- Each questionnaire has it's own folder, within this:
+  - A top-level questionnaire definition Jsonnet file 
+  - A `lib/` folder holds common Jsonnet used across multiple blocks
+    - The file extension should be `.libsonnet`
+    - Each file here can extend common Jsonnet from the questionnaire level lib folder
+    - `rules.libsonnet` extends the top level `lib/common_rules.libsonnet`
+  - A `blocks` folder:
+    - Subdivided into section folders
+    - Within each section folder a Jsonnet file per block
+  
+- We currently have the following questionnaires:
+
+    - `ccs`
+    - `communal-establishment`
+    - `household`
+    - `individual`
+
+- Blocks from the `individual` folder are also used from the `household` questionnaire.
+
+## Global variables
+
+- Global variables are variables known at questionnaire launch. They don't change during completion of a questionnaire.
+- Top-level arguments (`--tla-str`) are used to pass variables that are used in 5 blocks or less.
+- External variables (`--ext-str`) are used to pass variables that are used in a more than 5 blocks.
```

Comment:
This confused me for a second, at first I thought it was a limitation of these rather than our preference. I think it might be clearer if it read:

```suggestion
- Top-level arguments (`--tla-str`) should be used to pass variables that are used in 5 blocks or less.
- External variables (`--ext-str`) should be used to pass variables that are used in more than 5 blocks.
```
Thoughts?